### PR TITLE
EG-2654 - Ensure stack traces are printed to the logs in error reporting

### PR DESCRIPTION
**Pull request by JamesHR3**
Error reporting was unconditionally using the version of `logger.error` that does not take an exception parameter. In cases where the error object is a throwable (which is the majority of them), the stack trace should also be printed to the logs. This change ensures that this actually happens.

It also adds the exception message to the logged message for the catch all case for database exceptions.

### CODE REVIEW

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt [hunk 1]
```diff
@@ -24,6 +25,7 @@ class ErrorReporterImplTest {
 
     private val loggerMock = Mockito.mock(Logger::class.java).also {
         Mockito.`when`(it.error(anyString())).then { logs.addAll(it.arguments) }
+        Mockito.`when`(it.error(anyString(), any(Exception::class.java))).then { params -> logs.addAll(params.arguments) }
```
Comment (pnemeth) [inline common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:28]: can you use
logs.addAll(it.arguments)
here as well?

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt [hunk 2]
```diff
@@ -39,7 +41,8 @@ class ErrorReporterImplTest {
     private enum class TestErrors : ErrorCodes {
         CASE1,
         CASE2,
-        CASE_3;
+        CASE_3,
```
Comment (pnemeth) [inline common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:44]: very minor: any reason this one has an underscore while they others don't?

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt [hunk 3]
```diff
@@ -59,6 +62,11 @@ class ErrorReporterImplTest {
         override val parameters = listOf<Any>()
     }
 
+    private class TestError4(cause: Exception?) : Exception("This is test error 4", cause), ErrorCode<TestErrors> {
+        override val code = TestErrors.CASE4
+        override val parameters = listOf<Any>()
+    }
+
     private fun createReporterImpl(localeTag: String?) : ErrorReporterImpl {
```
Comment (pnemeth) [inline common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:70]: Seems like we always call this with a value, does localeTag have to be nullable?

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt (hunk 2 above)
Comment (pnemeth) [reply common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:44]: Sorry this is probably for the error code with underscore test

**Review by pnemeth — COMMENTED** [review]
LGTM

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt [hunk 4]
```diff
@@ -39,7 +41,8 @@ class ErrorReporterImplTest {
     private enum class TestErrors : ErrorCodes {
         CASE1,
         CASE2,
-        CASE_3;
+        CASE_3,
+        CASE4;
```
Comment (lankydan) [inline common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:45]: i dont really care, but why are these spelt differently (some have "_")

**Review by lankydan — APPROVED** [review]
Looks fine in general to me.

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt (hunk 4 above)
Comment (JamesHR3) [reply common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:45]: It's to catch an edge case, we do some _ to - replacement in a few places and I missed one.

File: common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt (hunk 1 above)
Comment (JamesHR3) [reply common/logging/src/test/kotlin/net/corda/commmon/logging/errorReporting/ErrorReporterImplTest.kt:28]: I probably should have changed both, but yes. As it's holding up a merge I think we'll just get it in for now and adjust it in the next change in 4.6.
